utils/src/test/kotlin/storage/FileArchiverTest.kt

```diff
@@ -22,53 +22,90 @@ package com.here.ort.utils.storage
 import com.here.ort.utils.safeMkdirs
 import com.here.ort.utils.unpack
 
+import io.kotlintest.TestCase
 import io.kotlintest.shouldBe
 import io.kotlintest.specs.StringSpec
 
-class FileArchiverTest : StringSpec({
-    "All files matching any of the patterns are archived" {
-        val dir = createTempDir("ort")
+import java.io.File
 
-        fun createFile(path: String) {
-            val file = dir.resolve(path)
-            file.parentFile.safeMkdirs()
-            file.writeText(path)
-        }
+class FileArchiverTest : StringSpec() {
+    lateinit var workingDir: File
+    lateinit var storageDir: File
+    lateinit var storage: LocalFileStorage
+
+    override fun beforeTest(testCase: TestCase) {
+        workingDir = createTempDir("ort").also { it.deleteOnExit() }
+        storageDir = createTempDir("ort").also { it.deleteOnExit() }
+        storage = LocalFileStorage(storageDir)
+    }
 
-        createFile("a")
-        createFile("b")
-        createFile("d/a")
-        createFile("d/b")
+    private fun createFile(path: String) {
+        val file = workingDir.resolve(path)
+        file.parentFile.safeMkdirs()
+        file.writeText(path)
+    }
 
-        val storageDir = createTempDir("ort")
-        val storage = LocalFileStorage(storageDir)
-        val archiver = FileArchiver(listOf("a", "**/a"), storage)
+    init {
+        "All files matching any of the patterns are archived" {
+            createFile("a")
+            createFile("b")
+            createFile("d/a")
+            createFile("d/b")
 
-        archiver.archive(dir, "save")
+            val archiver = FileArchiver(listOf("a", "**/a"), storage)
 
-        val archiveFile = storageDir.resolve("save/archive.zip")
+            archiver.archive(workingDir, "save")
 
-        archiveFile.isFile shouldBe true
+            val archiveFile = storageDir.resolve("save/archive.zip")
 
-        val unzipDir = createTempDir()
-        unzipDir.deleteOnExit()
-        archiveFile.unpack(unzipDir)
+            archiveFile.isFile shouldBe true
 
-        fun assertFileSaved(path: String) {
-            val file = unzipDir.resolve(path)
-            file.isFile shouldBe true
-            file.readText() shouldBe path
-        }
+            val unzipDir = createTempDir()
```

Comment:
Lines 61-71 are now duplicated in the new test case. If one would move the tempDir creation to `beforeTest`
and also move the `assertFileSaved()` to class level (and slightly change it) that redundancy was removed. Would something like that work out ?

Reply:
Done.